core/src/repair_service.rs

```diff
@@ -27,15 +27,23 @@ pub enum RepairType {
 struct RepairInfo {
     max_slot: u64,
     repair_tries: u64,
+    // List of (slot, weight) representing weighted forks
+    forks: Vec<(u64, u64)>,
 }
 
 impl RepairInfo {
     fn new() -> Self {
         RepairInfo {
             max_slot: 0,
             repair_tries: 0,
+            forks: vec![],
         }
     }
+
+    fn set_forks(&mut self, mut forks: Vec<(u64, u64)>) {
```

Comment:
This is the set of active slots that ReplayStage is working on, ordered by our lockout weighting scheme. @aeyakovenko, @rob-solana thoughts on how should this information be communicated to RepairService? A convenient solution would be to pass BankForks, but @rob-solana has brought up a good point that this might be a little heavy-handed/make testing harder. Maybe some API exposed through blocktree to notify listeners of forks replay_stage cares about, similar to the channel used to communicate updates to replay_stage?

Reply:
how about just an slots_of_interest: RwLock<Vec<u64>> in blocktree?

Reply:
when any consumer customers of Blocktree are trying to pull data out, they can add their interest to Blocktree.  Network observers and repair facilities can ask Blocktree "what are your customers interested in having repaired?"

Reply:
@sagar-solana

Reply:
@rob-solana, that works!